src/meal/models.py

```diff
@@ -114,12 +114,16 @@ class Component_ingredient(models.Model):
     component = models.ForeignKey(
         'meal.Component',
         verbose_name=_('component'),
-        related_name='+')
+        related_name='+',
+        on_delete=models.CASCADE
+    )
 
     ingredient = models.ForeignKey(
         'meal.Ingredient',
         verbose_name=_('ingredient'),
-        related_name='+')
+        related_name='+',
+        on_delete=models.CASCADE
```

Comment:
In fact what I have thought is wrong. We can have models.CASCADE here in m2m through models. In django admin, when we delete a m2m-related object, this will happen:
![image](https://cloud.githubusercontent.com/assets/8630726/23528875/5cc81d9e-ff69-11e6-92fb-ef5b7998343b.png)

So in fact, it will only affect this particular object that we're deleting. No problem with models.CASCADE.